source/common/upstream/cds_api_impl.cc

```diff
@@ -57,7 +57,16 @@ void CdsApiImpl::onConfigUpdate(const ResourceVector& resources, const std::stri
     const std::string cluster_name = cluster.name();
     try {
       clusters_to_remove.erase(cluster_name);
-      if (cm_.addOrUpdateCluster(cluster, version_info)) {
+      if (cm_.addOrUpdateCluster(cluster, version_info, [this](auto, auto state) {
```

Comment:
nit: please avoid auto here as it's hard to quickly read what the function is taking.